### ISSUE TITLE
ffmpeg libavutil common.h error missing -D__STDC_CONSTANT_MACROS

### ISSUE THREAD

**Issue by khelkun** [Streampunk/codecadon#2]
I'm using a Debian 7.11 (wheezy)

On "npm install codecadon" I had this error:
```
  CXX(target) Release/obj.target/codecadon/src/ScaleConverterFF.o
In file included from ../ffmpeg/include/libavutil/avutil.h:288:0,
                 from ../ffmpeg/include/libavutil/imgutils.h:30,
                 from ../src/ScaleConverterFF.cc:22:
../ffmpeg/include/libavutil/common.h:30:2: error: #error missing -D__STDC_CONSTANT_MACROS / #define __STDC_CONSTANT_MACROS
In file included from ../ffmpeg/include/libavutil/avutil.h:288:0,
                 from ../ffmpeg/include/libavutil/imgutils.h:30,
                 from ../src/ScaleConverterFF.cc:22:
../ffmpeg/include/libavutil/common.h: In function ‘int32_t av_clipl_int32_c(int64_t)’:
../ffmpeg/include/libavutil/common.h:205:47: error: ‘UINT64_C’ was not declared in this scope
make: *** [Release/obj.target/codecadon/src/ScaleConverterFF.o] Error 1
```
So I cloned codecadon repository and added "-D__STDC_CONSTANT_MACROS" to "cflags_cc" in binding.gyp
And the "npm install" did good. Not sure if this should be added to the repository or not.

N.B : I encountered the issue while trying to do Example 1 of node-red-contrib-dynamorse-core. I needed to install node-red-contrib-dynamorse-rtp-io to have pcap-reader node and it failed installing it while trying to install codecadon.
If you install codecadon manually like I did then install node-red-contrib-dynamorse-rtp-io you may get this npm error:
`npm ERR! git /home/khelkun/.node-red/node_modules/codecadon: Appears to be a git repo or submodule.`
Just rm -R /home/khelkun/.node-red/node_modules/codecadon/.git

**Comment by scriptorian** [comment]
Hi,
I've looked into this and there are a few people having the problem - I haven't understood why I didn't though. I am happy to add the define but looking at the binding.gyp docs they suggest a slightly different way of generating it. Would you be happy to test the following on your system as it isn't going wrong for me?
<snip>
              { # ia32 or x64
                "defines": [
                  "__STDC_CONSTANT_MACROS"
                ],
                "cflags_cc!": [
                  "-fno-rtti",
</snip>

Thanks!

**Comment by khelkun** [comment]
Hi 
Thanks for your answer and sorry I didn't answered earlier. I added:
```
"defines": [
"__STDC_CONSTANT_MACROS"
],
```
where you told me to and node-gyp builds fine with this additionnal macro definiton.
I should say again that I did that on a Debian 7. But unfortunatetly on Debian 7 I was unable to play with dynamorse examples because of GLIBC version requirement (Debian 7 GLIBC version is too old and I don't really want to risk updating it).
So I've setup a Debian 9 virtual machine and I'm currently playing with node-red on it. I've not tried dynamorse yet but I'm pretty sure it will be fine.
